anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/DetectorManager.java

```diff
@@ -37,13 +38,33 @@
 @RequiredArgsConstructor
 @Slf4j
 public class DetectorManager {
+    private static final String CK_DETECTOR_REFRESH_PERIOD = "detector-refresh-period";
+    private final ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
 
     @Getter
     @NonNull
     private DetectorSource detectorSource;
 
+    private int detectorRefreshTimePeriod;
+
     private final Map<UUID, AnomalyDetector> cachedDetectors = new HashMap<>();
 
+    public DetectorManager(DetectorSource detectorSource, Config config) {
+        this.detectorSource = detectorSource;
+        this.detectorRefreshTimePeriod = config.getInt(CK_DETECTOR_REFRESH_PERIOD);
+        this.initScheduler();
+    }
+
+    private void initScheduler() {
+        scheduler.scheduleWithFixedDelay(() -> {
+            try {
+                this.detectorMapRefresh();
+            } catch (Exception e) {
+                log.error("Error refreshing detectors", e);
+            }
+        }, 1, detectorRefreshTimePeriod, TimeUnit.MINUTES);
+    }
```

Comment:
If we get an exception, do we just lose the refresh until we restart the manager? What kind of exception are we expecting here, and is it something that we can recover from without losing refreshes entirely?

Reply:
This is to catch DetectorException's and any other Null pointer exceptions that might occur downstream. We don't loose the refresh when exception is caught.